internal/escape/escape.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package escape includes helpers for escaping and unescaping strings.
+package escape
+
+// Escape returns s, with all bytes for which shouldEscape returns true
+// escaped to "__0xXX__", where XX is the hex representation of the byte
+// value.
+//
+// shouldEscape takes the whole string and an index instead of a single
+// byte because some escape decisions require context. For example, we
+// might want to escape the second "/" in "//" but not the first one.
+func Escape(shouldEscape func(s string, i int) bool, s string) string {
```

Comment:
Put `s` first. It will make it easier to read calls that inline the shouldEscape function.

Reply:
Good point; done.